### DISCUSSION TITLE
StockPurchasable returns false despite item being in stock

### DISCUSSION THREAD

**Discussion by maplesyrupman** [swellstores/community#237] [Help] [answered]
When I query a product and include the `stockPurchasable` field, it returns `false` even though the `stockLevel` is 15 and `stockStatus` is 'in-stock' 

When I toggle `allow the product to be purchased when out of stock` on the admin dashboard, `stockPurchasable` then becomes `true`. 

Any help or explanation is greatly appreciated.

**Comment by sags95** [comment]
When `stock_purchasable` is set to `true` (through the toggle you mention), this indicates that the product can be purchase regardless of the `stock_status`. When it's set to `false`, this means an order will be blocked unless the `stock_status` is `in_stock`, `preorder`, or `backorder`.

**Reply by maplesyrupman** [reply]
I understand, that's not my problem. It should be returning true even when `allow the product to be purchased when out of stock` is toggled off because there are 15 items in stock.

**Reply by logeshswell** [reply, marked as answer]
@maplesyrupman Sorry for the confusion. `stock_purchasable` represents the toggle: 'allow the product to be purchased when out of stock'

If it's enabled, value will be `true`, else it will be set as `false`.
